Auth/app/src/main/java/com/example/Auth/BioActivity.kt

```diff
@@ -0,0 +1,316 @@
+package com.example.Auth
+
+
+import android.app.Activity
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.os.AsyncTask
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.Auth.databinding.ActivityBioBinding
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import okhttp3.*
+import java.io.ByteArrayOutputStream
+import java.util.*
+
+private val jsonFormat = Json {
+    coerceInputValues = true; ignoreUnknownKeys = true
+}
+
+class BioActivity : AppCompatActivity(), OnKnopkaClickListener {
+    lateinit var binding: ActivityBioBinding
+    private val adapter = KnopkaFeedAdapter(this)
+//    private var ind: Int = 0;
+
+    class MainActivityUnits {
+
+        var changeInfoButton: Button? = null
+        var textViewName: TextView? = null
+        var textViewBio: TextView? = null
+        var imageViewProfilePic: ImageView? = null
+        var googleLogOutButton: Button? = null
+        var profilePicBitMap: Bitmap? = null
+        var token: String? = null
+        var id: Int = 1
+    }
+
+    private var units: MainActivityUnits = MainActivityUnits()
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun bitMapToBase64String(bitMap: Bitmap?): String? {
```

Comment:
It is better when methods come after their usage in the same file like this:
fun doThing() {
   ...
   formatThing(...)
...
}
fun formatThing()  {...}